addon/search/search.js

```diff
@@ -158,6 +201,10 @@
     });
   }
 
+  CodeMirror.defineOption("incrementalSearch", false, function(cm, val) {
+    cm.state.incrementalSearch = (val === true);
```

Comment:
See above.  I can retry a command method implementation if you still think it makes sense.  I'm not sure if this is really an incremental vs non-incremental search thing.  It is pretty much the same search with different keybindings and the dialog stays open rather than closing.